File: testFixtures/src/main/java/de/medizininformatikinitiative/medgraph/TempDirectoryTestExtension.java
package de.medizininformatikinitiative.medgraph;

import org.junit.jupiter.api.extension.*;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Comparator;

/**
 * Test extension which supplies a temporary directory for each test. By specifying a {@link File} or {@link Path}
 * parameter on your test, setup, teardown or related methods, you can access this directory, as it will be injected
 * with this parameter.
 *
 * @author Markus Budeus
 */
public class TempDirectoryTestExtension implements Extension, BeforeEachCallback, AfterEachCallback, ParameterResolver {

	private static final ExtensionContext.Namespace NAMESPACE = ExtensionContext.Namespace.create(TempDirectoryTestExtension.class);
	private static final String DIRECTORY_KEY = "tempdir";
	private static final String DIRECTORY_NAME = "de.medizininformatikinitiative.medgraph.test";

	@Override
	public void beforeEach(ExtensionContext context) throws Exception {
		context.getStore(NAMESPACE).put(DIRECTORY_KEY, Files.createTempDirectory(DIRECTORY_NAME));
	}

	@Override
	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
	throws ParameterResolutionException {
		return parameterContext.getParameter().getType() == File.class
				|| parameterContext.getParameter().getType() == Path.class;
	}

	@Override
	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
	throws ParameterResolutionException {
		Path path = (Path) extensionContext.getStore(NAMESPACE).get(DIRECTORY_KEY);
		if (parameterContext.getParameter().getType() == File.class) return path.toFile();
		return path;
	}

	@Override
	public void afterEach(ExtensionContext context) throws Exception {
		ExtensionContext.Store store = context.getStore(NAMESPACE);
		Path dir = (Path) store.get(DIRECTORY_KEY);
		if (dir != null) {
			Files.walk(dir)
			     .sorted(Comparator.reverseOrder())
			     .map(Path::toFile)
			     .forEach(java.io.File::delete);

			store.remove(DIRECTORY_KEY);
		}
	}
}

File: composeApp/src/commonTest/kotlin/de/medizininformatikinitiative/medgraph/ui/TempDirectoryTestExtension.kt
package de.medizininformatikinitiative.medgraph.ui

import org.junit.jupiter.api.extension.*
import java.io.File
import java.nio.file.Files
import java.nio.file.Path

/**
 * Test extension which supplies a temporary directory for each test. The directory is injected as parameter.
 *
 * @author Markus Budeus
 */
class TempDirectoryTestExtension : Extension, BeforeEachCallback, AfterEachCallback, ParameterResolver {

    companion object {
        private val NAMESPACE = ExtensionContext.Namespace.create(TempDirectoryTestExtension::class.java)
        private val DIRECTORY_KEY = "tempdir"
        private val DIRECTORY_NAME = "de.medizininformatikinitiative.medgraph.test"
    }

    override fun beforeEach(context: ExtensionContext) {
        context.getStore(NAMESPACE)
            .put(DIRECTORY_KEY, Files.createTempDirectory(DIRECTORY_NAME))
    }

    override fun supportsParameter(parameterContext: ParameterContext, extensionContext: ExtensionContext): Boolean {
        return parameterContext.parameter.type == File::class.java
                || parameterContext.parameter.type == Path::class.java
    }

    override fun resolveParameter(parameterContext: ParameterContext, extensionContext: ExtensionContext): Any {
        val path: Path = extensionContext.getStore(NAMESPACE).get(DIRECTORY_KEY) as Path
        if (parameterContext.parameter.type == File::class.java) return path.toFile()
        return path
    }

    override fun afterEach(context: ExtensionContext) {
        val store = context.getStore(NAMESPACE)
        val dir: Path? = store.get(DIRECTORY_KEY) as? Path
        if (dir != null) {
            Files.walk(dir)
                .sorted(Comparator.reverseOrder())
                .map(Path::toFile)
                .forEach(File::delete)

            store.remove(DIRECTORY_KEY)
        }
    }

}